### PR TITLE
namespace: fall back gracefully when kernel doesn't support network namespaces

### PR DESCRIPTION
**Pull request by msekletar**

### CODE REVIEW

File: src/core/namespace.c [hunk 1]
```diff
@@ -1428,6 +1428,17 @@ int setup_netns(int netns_storage_socket[2]) {
         return r;
 }
 
+bool ns_type_supported(NameSpaceType type) {
+        const char *t, *ns_proc;
+
+        assert(type > _NAME_SPACE_TYPE_INVALID && type < _NAME_SPACE_TYPE_MAX);
```
Comment (poettering) [inline src/core/namespace.c]: maybe just return false here if the type is out of range? I mean, an unknown network namespace type is definitely not supported ;-)

**Comment by msekletar** [comment]
CI failure looks unrelated. @martinpitt any idea what is going on?

File: src/core/namespace.c (hunk 1 above)
Comment (msekletar) [reply src/core/namespace.c]: I think I have derived my spelling from ```NameSpaceInfo``` which also lives in ```namespace.h```.  I figure we should rename it to ```NamespaceInfo```.

**Comment by msekletar** [comment]
I force pushed my feature branch. Type is now named ```NamespaceType``` and ```ns_type_supported()``` returns ```false``` when argument is outside of supported range.

File: src/core/namespace.c (hunk 1 above)
Comment (poettering) [reply src/core/namespace.c]: yupp, we should

Comment (poettering) [reply src/core/namespace.c]: I did that now in PR #7045.

**Pull request merged**